[ADD] add module stock_backorder_strategy

laetitia-gangloff opened the pull request:
Define on stock_picking_type how to manage backorder.
create : default behavior
no_create : no backorder created
cancel : backorder is cancelled

yvaucher commented on stock_picking_backorder_strategy/README.rst:
```diff
@@ -0,0 +1,70 @@
+In types of operation add a backorder strategy to be able to cancel created
+backorder or to avoid to create a backorder.
+
+Possible strategies are :
```
Nitpicking of nitpicking: `:` doesn't have a space before it in english. In opposition to french for which it needs to be there.

Same for the 3 next lines

yvaucher commented:
:+1:

laetitia-gangloff commented:
What can I do for pylint ?

stock_picking_backorder_strategy/models/stock.py:31: [W0102(dangerous-default-value), StockPicking._create_backorder] Dangerous default value [] as argument (https://travis-ci.org/OCA/stock-logistics-workflow/jobs/75288135#L177)

yvaucher commented:
@laetitia-gangloff Well I checked it. It would have no side effect as if this argument `backorder_moves` gets it's default mutable value `[]` in official code the variable is check with `if not` and thus it set it to value of a field from picking.

I forgot pylint wouldn't let this pass, so I think you can simply put None instead of [] as default value.

If you want to do more, you can do as we usually did in old API with context.

laetitia-gangloff commented:
@yvaucher thank you !

I put None, and test it like with context in the old API.

pedrobaeza commented on stock_picking_backorder_strategy/__openerp__.py:
```diff
@@ -0,0 +1,36 @@
+    "name": "Picking backordering strategies",
+    "version": "0.1",
+    'author': "Acsone SA/NV, Odoo Community Association (OCA)",
+    "category": "Other",
```
Put "Warehouse Management"

pedrobaeza commented:
One last thing, but :+1:

Commit:
[ADD] add module stock_backorder_strategy

Define on stock_picking_type how to manage backorder.
create: default behavior
no_create: no backorder created
cancel: backorder is cancelled

lmignon commented:
:+1:  Thank you @laetitia-gangloff
